Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer+HighLowWatermarkBackPressureStrategy.swift

```diff
@@ -0,0 +1,46 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2022 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+/// A high-low watermarked back-pressure strategy for a ``NIOAsyncSequenceProducer``.
+///
+/// This strategy does the following:
+/// - On yield it keeps on demanding more as long as the `highWatermark` isn't reached.
+/// - On next it starts to demand once the the `lowWatermark` is reached.
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+public struct NIOAsyncSequenceProducerHighLowWatermarkBackPressureStrategy: NIOAsyncSequenceProducerBackPressureStrategy {
```

Comment:
I think we should have a namespace for these. Something like `enum NIOAsyncSequenceProducerStrategies`. That'll let us give this a better name.